src/js/extensions/anchor.js

```diff
@@ -60,10 +60,17 @@ var AnchorForm;
             event.preventDefault();
             event.stopPropagation();
 
-            var selectedParentElement = Selection.getSelectedParentElement(Selection.getSelectionRange(this.document)),
-                firstTextNode = Util.getFirstTextNode(selectedParentElement);
+            var linkSelected = false,
+                range = Selection.getSelectionRange(this.document);
+
+            if (range.startContainer.nodeName.toLowerCase() === 'a' ||
+                range.endContainer.nodeName.toLowerCase() === 'a') {
+                linkSelected = true;
+            } else if (Util.getClosestTag(Selection.getSelectedParentElement(range), 'a')) {
+                linkSelected = true;
+            }
```

Comment:
I was trying to make things more readable since these are 2 different kinds of checks, but I agree it'll be easier to just make it one if statement